### ISSUE TITLE
File upload UI mockup

### ISSUE THREAD

**Issue by CarlosNZ** [openmsupply/application-manager-web-app#431]
From File upload management [Epic #41](https://github.com/openmsupply/application-manager-web-app/issues/430)

Create wireframe for Form element plugin

**Comment by andreievg** [comment]
from @nmadruga on another issue: 

> Just one idea here from a website that O was using to add some docs uploaded (using mobile web site), could be used for reference...

> Before uploading file

![110858854-767c1400-831f-11eb-9874-7144886d943d (1)](https://user-images.githubusercontent.com/26194949/110861916-b34a0a00-8323-11eb-81db-1adaa1c2b7a0.png)
![110859002-a7f4df80-831f-11eb-946c-d07ede94a230](https://user-images.githubusercontent.com/26194949/110861928-b644fa80-8323-11eb-9d96-bdbfe99ac7f3.png)
![110859664-a2e46000-8320-11eb-8d34-613cc4e3d8d5](https://user-images.githubusercontent.com/26194949/110861946-ba711800-8323-11eb-84fd-67f7032fb72b.png)

**Comment by CarlosNZ** [comment]
Here is Deb's latest mockup:
![Screen Shot 2021-03-15 at 12 28 58 PM](https://user-images.githubusercontent.com/5456533/111089443-cad9fa80-8590-11eb-960e-0831643da5af.png)

Couple quick comments:
- I think we agreed not do do drag-n-drop for MVP, right?
- I don't know how we do progress bar. The endpoint only returns a response once upload is complete. From my basic research, it seems like tracking the amount of data sent so you can calculate progress is not something that can easily be done in a Fetch POST request. There's packages like axios that make this easier, but probably best not to worry about this for now, right? My suggestion for now would be to show the standard "Loading" spinner in the place that the thumbnails will appear.
- Really like the delete-on-hover icon, seems like a good way to do it without cluttering the UI.

With that in mind, would something like this do for first demo: ?

(Original at: https://balsamiq.cloud/scs7giw/ponj59g/r93F3)

**Initial view:**
![Screen Shot 2021-03-15 at 1 26 08 PM](https://user-images.githubusercontent.com/5456533/111089805-4daf8500-8592-11eb-89ad-f907d5122c1b.png)

**2 files uploaded, one more currently uploading:**
![Screen Shot 2021-03-15 at 1 27 52 PM](https://user-images.githubusercontent.com/5456533/111089823-599b4700-8592-11eb-967a-8f705c9d8eff.png)

**Files uploaded, hover to delete:**
![Screen Shot 2021-03-15 at 1 27 31 PM](https://user-images.githubusercontent.com/5456533/111089839-6324af00-8592-11eb-908f-3feba35cbd62.png)

Clicking either the thumbnail or the filename link loads the file in new tab (later iterations can have "slider box" overlays on current page, but keep it simple for first demo).

Summary View would just be the bottom part of the same UI (the files, without the interaction/button)

**Comment by andreievg** [comment]
> Really like the delete-on-hover icon, seems like a good way to do it without cluttering the UI.

Not really sure what you mean by this, since there is also this comment (`Clicking either the thumbnail or the filename link loads the file in new tab (later iterations can have "slider box" overlays on current page, but keep it simple for first demo`)


> I think we agreed not do do drag-n-drop for MVP, right?

For the first iteration yeah, but plugin can still match the designs

> I don't know how we do progress bar. The endpoint only returns a response once upload is complete. From my basic research, it seems like tracking the amount of data sent so you can calculate progress is not something that can easily be done in a Fetch POST request. There's packages like axios that make this easier, but probably best not to worry about this for now, right? My suggestion for now would be to show the standard "Loading" spinner in the place that the thumbnails will appear.

Your designs looks like a good option

>Clicking either the thumbnail or the filename link loads the file in new tab (later iterations can have "slider box" overlays on current page, but keep it simple for first demo).

Sweet

**Comment by CarlosNZ** [comment]
> > Really like the delete-on-hover icon, seems like a good way to do it without cluttering the UI.
> 
> Not really sure what you mean by this, since there is also this comment (`Clicking either the thumbnail or the filename link loads the file in new tab (later iterations can have "slider box" overlays on current page, but keep it simple for first demo`)

In my mockup and in Debs, there is a small "X" icon that only appears when you hover over the file icon. So you click the "X" to delete the file, but clicking anywhere else on it opens/downloads it. Same as your browser tabs -- click the "X" to close, or elsewhere on the tab to go to that tab:
![Screen Shot 2021-03-15 at 2 45 38 PM](https://user-images.githubusercontent.com/5456533/111093198-2c07cb00-859d-11eb-85b5-5cbd2f81a82c.png)

**Comment by CarlosNZ** [comment]
Additional refinements to be part of new issues in [Epic 41](https://github.com/openmsupply/application-manager-web-app/issues/430)
